src/krylov/gmres.c

```diff
@@ -373,13 +375,22 @@ hypre_GMRESSolve(void  *gmres_vdata,
 
    (*(gmres_functions->CopyVector))(b, p[0]);
 
-   /* compute initial residual */
+   /* Compute initial residual */
    (*(gmres_functions->Matvec))(matvec_data, -1.0, A, x, 1.0, p[0]);
 
    (*(gmres_functions->InnerProd))(b, b, &num_tags, &biprod);
    b_norm = hypre_sqrt(biprod[0]);
    real_r_norm_old = b_norm;
 
+   /* Compute initial error when a reference solution is provided */
+   if (xref)
+   {
+      (*(gmres_functions->CopyVector))(x, w_3);
+      (*(gmres_functions->Axpy))(-1.0, xref, w_3);
+      (*(gmres_functions->InnerProd))(w_3, w_3, &num_tags, &xiprod);
+      e_norm = hypre_sqrt(xiprod[0]);
+   }
```

Comment:
Should this also check for print_level > 6 before performing the computations (and printing the initial error info)?

Reply:
I think checking if `xref` is passed in should be enough here as we still want to see initial/final error norms when `print_level > 1`

Reply:
ok